Add an error instead of triggering warning

kamil-tekiela opened the pull request:
When in development mode and you enter wrong credentials, the error is displayed on a blank page. This is quite annoying and inconsistent with production. But if we add the error directly then it behaves the same in both modes. 

The line and file name doesn't make any difference in this case. 

@MauricioFauth If this makes sense, I would like to refactor more to tidy this up.

kamil-tekiela committed:
Add an error instead of triggering warning

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

MauricioFauth reviewed (approved):
That's great! Is it possible to replace all trigger_error with that?

kamil-tekiela commented:
> That's great! Is it possible to replace all trigger_error with that?

I'm going to check now.

MauricioFauth commented:
> I'm going to check now.

I can't test it right now, but if it's possible, that would be great.

kamil-tekiela committed:
Refactor AuthenticationConfig

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela commented:
I found two more. We go from this:
![image](https://github.com/user-attachments/assets/a0f0f276-d4da-40ba-9993-4adfc1243972)
to this:
![image](https://github.com/user-attachments/assets/5129782a-15f4-4781-ab22-967122f499c1)
And this is how it looks like in development mode before this PR:
![image](https://github.com/user-attachments/assets/de94e668-ae07-4181-9290-933546d9ff9a)

kamil-tekiela committed:
Replace trigger_error with addError

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela committed:
Use new mysqli instead of mysqli_init

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela committed:
Create addNotice and addUserError

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela committed:
Don't display Backtrace if there isn't one

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela committed:
Convert more trigger_error

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela committed:
Replace E_USER_ERROR with exception

Both will stop processing the current code.

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela committed:
Refactor handleError

We no longer expect any E_USER_* errors to be triggered by PMA code.

Signed-off-by: Kamil Tekiela <tekiela246@gmail.com>

kamil-tekiela commented:
So I went ahead and removed all but one of `trigger_error`. 

- E_USER_WARNING is now converted to `addUserError`. It no longer calls the default PHP error handler.
- E_USER_NOTICE is now converted to `addNotice`. It no longer calls the default PHP error handler.
- E_USER_ERROR was replaced with exceptions. Since we do not handle errors in the error handler, this error would stop the script execution so it is kind of pointless to use a `trigger_error` for something that can be done more easily with an exception and achieve the exact same effect.
- E_USER_DEPRECATED remains because I don't know how to handle it. It's probably the least problematic of all. It's handled in a slightly strange way but it's used in only one place.

The next step is to improve the error handling code.

williamdes reviewed (approved):
Good idea !

MauricioFauth reviewed (approved):
Thank you!
